### PR TITLE
Fix automatic compaction and clarify Console context spend

### PR DESCRIPTION
**Pull request by rmusser01**
## Summary

- preserve deliberate per-chat Automatic compaction choices while keeping untouched inherited settings sparse
- show context fullness alongside completed **Current** spend and estimated incremental **On next send** additional charge
- avoid double-counting a locally estimated user turn when the provider reports actual usage for the following assistant response
- keep spend/context estimates accurate across dispatch, failure, continuation, remote recovery, media, cache, validation, and pricing edge cases
- provide responsive full and compact labels without clipping at production terminal sizes

## Verification

- 204 focused Console cost, status-chip, wiring, projection, rail, and mounted-screen tests passed after rebasing onto dev at 68f9d865fa
- additional context-control suite matched clean `dev` exactly: 25 passed and the same 6 pre-existing failures
- Ruff lint and format checks passed across all touched Python files
- `py_compile` passed for touched production modules
- generated CSS bundles are synchronized
- `git diff --check` passed
- all six Qodo findings addressed: canonical admitted-media filtering, persisted recovery IDs, API documentation, and safe submission IDs in error logs
- reviewed diagnostic statement changes and regenerated the diagnostic inventory; no new sinks or user-content logging
- fixed the startup regression from eager media configuration: the UI-ready census and media regressions now pass (10 tests), with 972 modules at startup and the existing limit unchanged

The full repository test suite was not run, following the repository policy requiring explicit opt-in.

### CODE REVIEW

File: tldw_chatbook/UI/Console_Modules/console_spend_projection.py [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+    {
+        ConsoleDispatchRecoveryKind.REMOTE_ACCEPTED,
+        ConsoleDispatchRecoveryKind.REMOTE_DISPATCH_STARTED,
+    }
+)
+
+
+def fold_system_prompt(system_prompt: str | None, greeting: str) -> str:
+    """Fold the seeded greeting exactly as the provider send path does."""
+    return fold_greeting_into_system_prompt(system_prompt or "", greeting)
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply tldw_chatbook/UI/Console_Modules/console_spend_projection.py:73]: Fixed in f6ca582f48: documented Args and Returns for all public spend builders, plus run_active and the refresh lifecycle methods.

File: tldw_chatbook/Widgets/Console/console_context_controls.py [hunk 2]
```diff
@@ -89,6 +92,62 @@ def conversation_row(self) -> str:
...
+
+
+def build_console_next_send_spend_state(
+    *,
+    request_tokens: int | None,
+    input_per_mtok: float | None,
+    sendable_text: bool,
+    has_media: bool,
+) -> ConsoleNextSendSpendState:
+    """Build an honest uncached-input forecast for the next request."""
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply tldw_chatbook/Widgets/Console/console_context_controls.py]: Fixed in f6ca582f48: both context/spend presentation builders now document every parameter and their return state.

File: tldw_chatbook/Widgets/Console/console_status_chips.py [hunk 3]
```diff
@@ -835,20 +839,32 @@ def sync_cost_state(self, state: ConsoleCostState | None) -> None:
...
+        if state is None:
+            return
+        # The approved breakpoint follows terminal width; the strip itself is
+        # two cells narrower because its production CSS owns horizontal padding.
+        label = state.compact_label if self.screen.size.width < 120 else state.label
+        chip.update(label)
+        chip.refresh(layout=True)
+
+    def on_resize(self, _event: events.Resize) -> None:
+        """Reapply width-aware cost copy without requiring a state change."""
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply tldw_chatbook/Widgets/Console/console_status_chips.py]: Fixed in f6ca582f48: on_resize now documents the Textual _event parameter.

File: tldw_chatbook/UI/Screens/chat_screen.py [hunk 4]
```diff
@@ -3557,55 +3977,571 @@ async def action_open_console_model_popover(self) -> None:
...
+            try:
+                _session, display_name_plan = (
+                    store.prepare_session_user_display_name_override_for_commit(
+                        committed.live_commit,
+                        submission.user_display_name_override,
+                        global_default=self._global_chat_display_name(),
+                    )
+                )
+            except Exception:
+                logger.exception("Console settings display-name preparation failed")
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply tldw_chatbook/UI/Screens/chat_screen.py]: Fixed in f6ca582f48: both exception logs include submission.submission_id. Reviewed the diagnostic statements and regenerated the inventory; no user text or settings values were added to logs.

File: tldw_chatbook/UI/Console_Modules/console_spend_projection.py [hunk 5]
```diff
@@ -0,0 +1,318 @@
...
+    current_ids: set[str] = set()
+    seen_user = False
+    for message in messages:
+        if message.role not in {
+            ConsoleMessageRole.USER,
+            ConsoleMessageRole.ASSISTANT,
+        }:
+            continue
+        excluded_from_request = message.id == request_excluded_user_id
+        excluded_from_current = message.id == current_excluded_user_id
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply tldw_chatbook/UI/Console_Modules/console_spend_projection.py]: Fixed in f6ca582f48: resolve checkpoint persisted user IDs to transient transcript IDs, retaining transient-ID fallback for ephemeral/preparation ownership. Distinct-ID regressions verify accepted users remain request context only and quarantined users are excluded from both projections. All 204 focused tests pass.

File: tldw_chatbook/UI/Console_Modules/console_spend_projection.py [hunk 6]
```diff
@@ -0,0 +1,318 @@
...
+    messages: Sequence[ConsoleChatMessage],
+    has_pending_attachments: bool,
+    request_tokens: int | None,
+    input_per_mtok: float | None,
+    draft_text: str,
+) -> ConsoleNextSendSpendState:
+    """Build the input-only forecast from text and attachment metadata."""
+    has_media = has_pending_attachments or any(
+        message.attachments or message.image_data is not None for message in messages
+    )
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply tldw_chatbook/UI/Console_Modules/console_spend_projection.py]: Fixed in f6ca582f48: media detection uses canonical metadata-only provider rows after lifecycle filtering, with usable-data, role, vision, and image-budget admission. Mounted regressions verify failed echoes, assistant attachments, missing data, non-vision models, and unaccepted turns retain a priced forecast. Provider serialization is forbidden in those tests. All 204 focused tests pass.

**Comment by rmusser01** [comment]
Rebased onto dev at 68f9d865fa and addressed all six Qodo findings in f6ca582f48.

- Historical media now uses the canonical metadata-only provider admission path after lifecycle filtering, including usable attachment data, role, vision capability, and image budgeting. Mounted regressions cover failed echoes, assistant media, missing bytes, non-vision requests, and unaccepted turns without serializing attachments.
- Recovery checkpoint exclusions resolve persisted user IDs to transient transcript IDs, preserving transient preparation ownership. Regressions use distinct database/transient IDs for accepted and quarantined recovery.
- Added the requested Google-style API and resize-event documentation.
- Both display-name error logs now include the safe submission ID. Reviewed the diagnostic statement diff (two identifier additions and one formatting-only change), then regenerated the inventory; no new sinks or user-content logging.

Verification: 204 focused tests passed; Ruff lint/format, compilation, and CSS reproduction passed. The additional context-control suite exactly matches clean dev: 25 passed and the same six pre-existing failures. No full-suite run. ADR required: no; these fixes implement the existing request-admission and recovery-ownership contracts.

**Comment by rmusser01** [comment]
Fixed the UI latency check failure in 24bffc3f40. Send-configuration capture was running on empty/text-only cost refreshes and importing the RAG stack before first paint (1,030 modules). It now runs only when lifecycle-admitted user attachments need capability/budget filtering. The existing UI-ready census passes at 972/972, and all 10 startup/media regressions pass. The ratchet limit is unchanged.
